pkg/endpoint/bpf.go

```diff
@@ -305,10 +306,11 @@ func writeGeneve(prefix string, e *Endpoint) ([]byte, error) {
 	return rawData, nil
 }
 
-func (e *Endpoint) runInit(libdir, rundir, epdir, debug string) error {
-	e.Mutex.RLock()
-	args := []string{libdir, rundir, epdir, e.IfName, debug}
+func (e *Endpoint) runInit(libdir, rundir, epdir, ifName, debug string) error {
+	args := []string{libdir, rundir, epdir, ifName, debug}
```

Comment:
endpoint's IfName can be set with the API "PATCH /endpoint/{id}" request. Not sure if we support changing it after it is set for the first time, but I presume it could be un-set after the endpoint is created?